arrow-libs/core/arrow-core-high-arity/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -17,6 +21,7 @@ public inline fun <T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, R> Iterable<T1>
   t11: Iterable<T11>,
   transform: (T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11) -> R,
 ): List<R> {
+  contract { callsInPlace(transform, InvocationKind.UNKNOWN) }
```

Comment:
Does this have any benefit with respect to not having any contract?

Reply:
Removed. It was useful because the compiler would report `LEAKED_IN_PLACE_LAMBDA` in the NonEmptyList version. However, that seems to just be a compiler bug since it can be gotten around by not passing the function explicitly (e.g. compiler would complain if one calls `map(f)` but not on `map { f(it) }`)